hpx/hpx_init_params.hpp

```diff
@@ -0,0 +1,85 @@
+//  Copyright (c)      2020 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file hpx_init_params.hpp
+
+#ifndef HPX_HPX_INIT_PARAMS_HPP
+#define HPX_HPX_INIT_PARAMS_HPP
+
+#include <hpx/config.hpp>
+#include <hpx/functional/function.hpp>
+#include <hpx/program_options.hpp>
+#include <hpx/resource_partitioner/partitioner_fwd.hpp>
+#include <hpx/runtime/runtime_mode.hpp>
+#include <hpx/runtime/shutdown_function.hpp>
+#include <hpx/runtime/startup_function.hpp>
+
+#include <functional>
+#include <string>
+#include <vector>
+
+#ifndef DOXYGEN
+///////////////////////////////////////////////////////////////////////////////
+// One of these functions must be implemented by the application for the
+// console locality.
+int hpx_main();
+int hpx_main(int argc, char** argv);
+int hpx_main(hpx::program_options::variables_map& vm);
+#endif
+
+///////////////////////////////////////////////////////////////////////////////
+/// \cond NOINTERNAL
+namespace hpx_startup
+{
+    // As an alternative, the user can provide a function hpx_startup::user_main,
+    // which is semantically equivalent to the plain old C-main.
+    int user_main();
+    int user_main(int argc, char** argv);
+}
+/// \endcond
+
+///////////////////////////////////////////////////////////////////////////////
+/// \namespace hpx
+namespace hpx
+{
+    namespace detail
+    {
+        HPX_EXPORT void on_exit() noexcept;
+        HPX_EXPORT void on_abort(int signal) noexcept;
+        // Default params to initialize the init_params struct
+        static char app_name[] = HPX_APPLICATION_STRING;
+        static const hpx::program_options::options_description default_desc =
+            hpx::program_options::options_description(
+                "Usage: " HPX_APPLICATION_STRING " [options]");
+        static startup_function_type default_startup = startup_function_type();
+        static shutdown_function_type default_shutdown = shutdown_function_type();
+    }
+
+#ifndef DOXYGEN
+    typedef int (*hpx_main_type)(hpx::program_options::variables_map&);
+    typedef int (*hpx_user_main_type)(int argc, char** argv);
+#endif
+
+    /// \struct init_params
+    /// \brief A struct to contain the hpx::init() parameters
+    struct init_params {
+        char *default_argv[2] = { detail::app_name , nullptr };
+        // Parameters
+        std::reference_wrapper<hpx::program_options::options_description const>
+            desc_cmdline = detail::default_desc;
+        int dummy_argc = 1;
+        char **dummy_argv = default_argv;
+        std::vector<std::string> cfg;
```

Comment:
```suggestion
        /// You can add member variable documentation like this.
        std::vector<std::string> cfg;
```

Reply:
Can I also put it like `\var <var_name>  <explanations>` on the very top of the struct ? Cause I feel like otherwise the code is more difficult to read